engine/src/main/java/org/pentaho/di/trans/steps/script/ScriptAddedFunctions.java

```diff
@@ -0,0 +1,2532 @@
+          return (String) undefinedValue;
+        }
+        String fillChar = (String) ArgList[0];
+        int count = (Integer) ArgList[1];
+        if ( fillChar.length() != 1 ) {
+          throw new RuntimeException( "Please provide a valid Char to the fillString" );
+        } else {
+          char[] chars = new char[count];
+          while ( count > 0 ) {
+            chars[--count] = fillChar.charAt( 0 );
+          }
+          return new String( chars );
+        }
+      } catch ( Exception e ) {
+        throw new RuntimeException( e.toString() );
+      }
+    } else {
+      throw new RuntimeException( "The function call fillString requires 2 arguments." );
+    }
+  }
+
+  public static Object isCodepage( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    boolean bRC = false;
+    if ( ArgList.length == 2 ) {
+      try {
+        if ( isNull( ArgList, new int[] { 0, 1 } ) ) {
+          return null;
+        } else if ( isUndefined( ArgList, new int[] { 0, 1 } ) ) {
+          return undefinedValue;
+        }
+        String strValueToCheck = (String) ArgList[0];
+        String strCodePage = (String) ArgList[1];
+        byte[] bytearray = strValueToCheck.getBytes();
+        CharsetDecoder d = Charset.forName( strCodePage ).newDecoder();
+        CharBuffer r = d.decode( ByteBuffer.wrap( bytearray ) );
+        r.toString();
+        bRC = true;
+      } catch ( Exception e ) {
+        bRC = false;
+      }
+    } else {
+      throw new RuntimeException( "The function call isCodepage requires 2 arguments." );
+    }
+    return Boolean.valueOf( bRC );
+  }
+
+  public static String ltrim( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    try {
+      if ( ArgList.length == 1 ) {
+        if ( isNull( ArgList[0] ) ) {
+          return null;
+        } else if ( isUndefined( ArgList[0] ) ) {
+          return (String) undefinedValue;
+        }
+        String strValueToTrim = (String) ArgList[0];
+        return strValueToTrim.replaceAll( "^\\s+", "" );
+      } else {
+        throw new RuntimeException( "The function call ltrim requires 1 argument." );
+      }
+    } catch ( Exception e ) {
+      throw new RuntimeException( "The function call ltrim is not valid : " + e.getMessage() );
+    }
+  }
+
+  public static String rtrim( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    try {
+      if ( ArgList.length == 1 ) {
+        if ( isNull( ArgList[0] ) ) {
+          return null;
+        } else if ( isUndefined( ArgList[0] ) ) {
+          return (String) undefinedValue;
+        }
+        String strValueToTrim = (String) ArgList[0];
+        return strValueToTrim.replaceAll( "\\s+$", "" );
+      } else {
+        throw new RuntimeException( "The function call rtrim requires 1 argument." );
+      }
+    } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)